### ISSUE TITLE
Allow progression only when all monsters have been killed, and all barrels destroyed

### ISSUE THREAD

**Issue by sixcy** [sixcy/devilutionX-ironman#5]

**Comment by sixcy** [comment]
Note that the TP from Lazarus (text "Portal to level 15") to Diablo is not handled yet by the renaming

**Comment by sixcy** [comment]
Most of it works! Few things need checking out or reworking:
- chests/sarcophagus and barrels could be clamped into one term: "objects"
- stairs to quest levels (like PWATER) are not really handled yet

**Comment by sixcy** [comment]
One small detail but it would be better if solved.. I cannot seem to be able to print more than 3 lines in the InfoString. I could look at fixing that

**Comment by sixcy** [comment]
Stairs to quest levels seems to be handled, but I have not tested extensively yet.

About clamping chests/sarcophagus/barrels into "objects", I think I prefer having barrels counted separately.
However, I can rename "objects" into "chests" to prevent confusion

**Comment by sixcy** [comment]
Done everything: I only need to test now to check I have not forgotten anything

**Comment by sixcy** [comment]
The handling of quest zones is a little buggy - it is possible to softlock by going to Chamber of Bones, clearing the level, going back to dlvl6, then back to Chamber of Bones. The chest counter becomes then bogus. Also, oddly enough, the counters are not displayed on the automap, specifically in the Chamber of Bones.

I will have to look into this eventually. For now, a potential bypass would consist in checking whether the quest is done: if the quest is done, disallow backtracking through the quest area. I think this bypass would be fine in terms of ironman rules, although I would have preferred (assuming all monsters killed etc..) free navigation between the dlvl and the associated quest area.

**Comment by sixcy** [comment]
I fixed the automap counter issue.

I tried the bypass (disallowing to go in when the quest is done) but there remains a possibility of softlock for Chamber of Bones if the player clears the level but does not read the guardian spell (thus the quest stays active).

I could try to mess around and force the player to read the spell to go back to dlvl6, but now this would just be spaghetti.

I will try diving deeper to get a solution to this chest count problem.
